src/__tests__/cucumber_tests/shared_steps/aquaSlider.ts

```diff
@@ -1,46 +1,362 @@
+    async (frequency: number, quality: string) => {
+      await setFrequencyQuality(webdriver, frequency, quality);
+    }
+  );
+};
+
+export const whenSetFrequencyQuality = (
+  when: DefineStepFunction,
+  webdriver: { driver: Driver | undefined }
+) => {
+  when(
+    /^I set the quality to (\d+(?:.\d+)?) for the band with frequency (\d+)Hz$/,
+    async (quality: string, frequency: number) => {
+      await setFrequencyQuality(webdriver, frequency, quality);
+    }
+  );
+};
+
+export const whenSetFrequencyQualityUsingArrows = (
+  when: DefineStepFunction,
+  webdriver: { driver: Driver | undefined }
+) => {
+  when(
+    /^I click on the (up|down) arrow for the quality for frequency (\d+)Hz (\d+) times$/,
+    async (direction: string, frequency: number, times: number) => {
+      const element = await webdriver.driver.$(
+        `.mainContent label[for="${frequency}-quality"]`
+      );
+      const button = await element.$(`.arrow-${direction}`);
+      for (let i = 0; i < times; i += 1) {
+        await button.click();
+        // wait 500 ms for the action. necessary
+        await new Promise((resolve) => setTimeout(resolve, 500));
       }
-      throw new Error(`${frequency} Hz gain band not found.`);
     }
   );
 };
 
-export const givenSetFrequencyGain = (
+// ====================================
+
+export const givenFrequencyFilterType = (
   given: DefineStepFunction,
   webdriver: { driver: Driver | undefined }
 ) => {
   given(
-    /^I set gain of slider of frequency (\d+)Hz to (top|middle|bottom)$/,
-    async (frequency: number, position: string) => {
-      console.log(`frequency: ${frequency}`);
-      console.log(`position: ${position}`);
-      // TODO use webdriver io get slider position
-      console.log(webdriver.driver);
+    /^the filter type is (\w+) filter for the band with frequency (\d+)Hz$/,
+    async (filterType: string, frequency: number) => {
+      if (
+        Object.values(FilterTypeEnum).findIndex((f) => f === filterType) === -1
+      ) {
+        throw new Error(`Invalid filter type ${filterType}.`);
+      }
+      const filterTypeAsEnum = filterType as FilterTypeEnum;
+      const dropdownElem = await webdriver.driver
+        .$('.mainContent')
+        .$(`.dropdown [aria-label="${frequency}-filter-type"]`);
+
+      expect(dropdownElem).not.toBeNull();
+      dropdownElem.click();
+      // wait 1000 ms for the action.
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+
+      // Need to reselect from driver since these elements didn't exist before clicking on the dropdown
+      const filterElement = await webdriver.driver.$(
+        `.dropdown li[aria-label="${FilterTypeToLabelMap[filterTypeAsEnum]}"]`
+      );
+      expect(filterElement).not.toBeNull();
+      filterElement.click();
+      // wait 1000 ms for the action.
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+    }
+  );
+};
+
+export const whenSetFrequencyFilterType = (
```

Comment:
We can refactor the code in this function and `givenFrequencyFilterType` into a `setFrequencyFilterType` function similar to how you handled the other test cases. It's probably more readable that way